src/index.js

```diff
@@ -1,9 +1,73 @@
 'use strict';
 
-const defaultOptions = {};
+const Matrix = require('ml-matrix');
+const Algebra = Matrix.algebra;
 
 function simulatedAnnealing(options) {
-    options = Object.assign({}, defaultOptions, options);
+    var f = options[0]; // Objective function
+    var x0 = options[1]; //Initial guess
+    var l = options[2]; // Lower bound
+    var u = options[3]; // Upper bound
```

Comment:
you should consider to rename to a more explicit variable name, instead of `u` rename it as `upperBound`, this applies for all the others